src/megatron/bridge/training/eval.py

```diff
@@ -99,15 +101,36 @@ def evaluate(
             if verbose:
                 print_rank_0(f"Evaluating iter {iteration}/{state.cfg.train.eval_iters}")
 
+            # Handle finetuning vs pretraining data consumption
+            seq_length = state.cfg.model.seq_length  # Default for pretraining
+            eval_data_iterator = data_iterator  # Default for pretraining
+
+            if state.cfg.dataset.dataloader_type == "batch":
```

Comment:
is this check to decide finetune a bit implicit

Reply:
it's tied to the sampler used, specifically for parity with this from nemo since we use the same global batch sampler: https://github.com/NVIDIA-NeMo/NeMo/blob/10327b8e58f300d0af40c90b22c0641412e951c5/nemo/lightning/megatron_parallel.py#L1481-L1497